ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLAgentExecutor.java

```diff
@@ -113,9 +116,14 @@ public void execute(Input input, ActionListener<Output> listener) {
                             MLMemorySpec memorySpec = mlAgent.getMemory();
                             String memoryId = inputDataSet.getParameters().get(MEMORY_ID);
                             String parentInteractionId = inputDataSet.getParameters().get(PARENT_INTERACTION_ID);
+                            String regenerateInteractionId = inputDataSet.getParameters().get(REGENERATE_INTERACTION_ID);
                             String appType = mlAgent.getAppType();
                             String question = inputDataSet.getParameters().get(QUESTION);
 
+                            if (memoryId == null && regenerateInteractionId != null) {
```

Comment:
What if both are null? Do we still want to continue?

Reply:
if `regenerateInteractionId` is null, it's a normal flow, memory will automatically create if needed